### PR TITLE
Requests Admin Page

### PR DESCRIPTION
**Pull request by ntdat213**

### CODE REVIEW

File: src/app/pages/admin-request/components/admin-request.component.scss [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    margin-top: 1.5em;
+    margin-bottom: 1.5em;
+}
+
+.bo-tron {
+    border-top: 2px solid #0000006b;
+    border-bottom: 2px solid #0000006b;
+    border-right: 2px solid #0000006b;
+    // border-top-right-radius: 20px;
+    // border-bottom-right-radius: 20px;
```
Comment (thanglenova) [inline src/app/pages/admin-request/components/admin-request.component.scss]: Please check and remove comment code

File: src/app/pages/admin-request/components/admin-request.component.ts [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+  putRequest(request: RequestModel) {
+    this.adminRequestService.putRequest(request);
+  }
+
+  onChange(event, request: RequestModel) {
+    this.httpClient.put("http://localhost:8081/api/requests", request).subscribe(pu => console.log(pu));
+    // console.log(request)
+    // this.putRequest(request);
+    // console.log(3);
```
Comment (thanglenova) [inline src/app/pages/admin-request/components/admin-request.component.ts]: Please check and remove comment code

File: src/app/pages/admin-request/services/request-type/request-type.service.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+import { RequestModel } from 'src/app/shared/models/request/request';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class RequestTypeService {
+
+  // private url = 'http://localhost:8081/api/request-types'
+
+  // constructor(private httpClient: HttpClient) { }
```
Comment (thanglenova) [inline src/app/pages/admin-request/services/request-type/request-type.service.ts]: Please check and remove comment code

File: src/app/pages/admin-request/components/admin-request.component.scss (hunk 1 above)
Comment (ntdat213) [reply src/app/pages/admin-request/components/admin-request.component.scss]: Done

File: src/app/pages/admin-request/components/admin-request.component.ts (hunk 2 above)
Comment (ntdat213) [reply src/app/pages/admin-request/components/admin-request.component.ts]: Done

File: src/app/pages/admin-request/services/request-type/request-type.service.ts (hunk 3 above)
Comment (ntdat213) [reply src/app/pages/admin-request/services/request-type/request-type.service.ts]: Done

File: src/app/pages/admin-request/components/admin-request.component.ts [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+  }
+
+  getSize() {
+    this.adminRequestService.getSize(this.searchKeyword).subscribe(size => {
+      this.sizeArray = size;
+      console.log(size);
```
[the remark this answers is not in the document]
Comment (ntdat213) [reply src/app/pages/admin-request/components/admin-request.component.ts]: Done

File: src/app/pages/admin-request/components/admin-request.component.ts [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+    });
+  }
+
+  requestPage(event) {
+    console.log(event);
+
```
[the remark this answers is not in the document]
Comment (ntdat213) [reply src/app/pages/admin-request/components/admin-request.component.ts]: Done

File: src/app/pages/admin-request/components/admin-request.component.ts [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+      items: items,
+      sortBy: sortBy,
+      search: search
+    }
+
+    this.adminRequestService.getPageRequest(body).subscribe(request => {
+      this.requests = request;
+      this.getSize();
+      console.log(body);
+    });
```
[the remark this answers is not in the document]
Comment (ntdat213) [reply src/app/pages/admin-request/components/admin-request.component.ts]: Done

File: src/app/pages/admin-request/components/admin-request.component.ts [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+  }
+
+  changeSortBy(event){
+
+    this.selectedSortBy = event.value;
+
+    this.sendRequestPage(0, this.selectedRows, this.selectedSortBy, this.searchKeyword);
```
[the remark this answers is not in the document]
Comment (ntdat213) [reply src/app/pages/admin-request/components/admin-request.component.ts]: Done

File: src/app/pages/admin-request/services/admin-request/admin-request.service.ts [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+  getPageRequest(body):Observable<RequestModel[]>{
+    let params = new HttpParams()
+    .append('page', body.page)
+    .append('items', body.items)
+    .append('sortBy', body.sortBy)
+    .append('search', body.search);
+
+    return this.httpClient.get<RequestModel[]>(this.url + '/pagination-and-search', {params:params}).pipe();
+  }
```
[the remark this answers is not in the document]
Comment (ntdat213) [reply src/app/pages/admin-request/services/admin-request/admin-request.service.ts]: Done

File: src/app/pages/admin-request/services/request-type/request-type.service.ts [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+
+
```
[the remark this answers is not in the document]
Comment (ntdat213) [reply src/app/pages/admin-request/services/request-type/request-type.service.ts]: Done

File: src/app/pages/admin-request/components/admin-request.component.ts [hunk 10]
```diff
@@ -0,0 +1,129 @@
...
+
+  sendRequestPage(page: number, items: number, sortBy: string, search: string) {
+
+    const body = {
+      page: page,
+      items: items,
+      sortBy: sortBy,
+      search: search
+    }
+
```
[the remark this answers is not in the document]
Comment (ntdat213) [reply src/app/pages/admin-request/components/admin-request.component.ts]: Done

File: src/app/pages/admin-request/services/admin-request/admin-request.service.spec.ts [hunk 11]
```diff
@@ -0,0 +1,12 @@
...
+import { AdminRequestService } from './admin-request.service';
+
+describe('AdminRequestService', () => {
+  beforeEach(() => TestBed.configureTestingModule({}));
+
+  it('should be created', () => {
+    const service: AdminRequestService = TestBed.get(AdminRequestService);
+    expect(service).toBeTruthy();
+  });
+});
```
[the remark this answers is not in the document]
Comment (ntdat213) [reply src/app/pages/admin-request/services/admin-request/admin-request.service.spec.ts]: Done

File: src/app/pages/admin-request/services/admin-request/admin-request.service.ts [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+import { RequestModel } from '../../../../shared/models/request/request';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AdminRequestService {
+
+  httpOptions = {
+    headers: new HttpHeaders({ 'Content-Type': 'application/json' })
+  };
```
[the remark this answers is not in the document]
Comment (ntdat213) [reply src/app/pages/admin-request/services/admin-request/admin-request.service.ts]: Done

File: src/app/pages/admin-request/services/request-type/request-type.service.spec.ts [hunk 13]
```diff
@@ -0,0 +1,12 @@
...
+import { RequestTypeService } from './request-type.service';
+
+describe('RequestTypeService', () => {
+  beforeEach(() => TestBed.configureTestingModule({}));
+
+  it('should be created', () => {
+    const service: RequestTypeService = TestBed.get(RequestTypeService);
+    expect(service).toBeTruthy();
+  });
+});
```
[the remark this answers is not in the document]
Comment (ntdat213) [reply src/app/pages/admin-request/services/request-type/request-type.service.spec.ts]: Done

File: src/app/pages/admin-request/services/status/status.service.ts [hunk 14]
```diff
@@ -0,0 +1,18 @@
...
+import { Observable } from 'rxjs';
+import { Status } from 'src/app/shared/models/request/status';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class StatusService {
+
+  private url = 'http://localhost:8081/api/status'
+
```
[the remark this answers is not in the document]
Comment (ntdat213) [reply src/app/pages/admin-request/services/status/status.service.ts]: Done

**Pull request merged**